sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+
+#include "byteorder.h"
+#include "cpu-conf.h"
+#include "kernel_types.h"
+#include "net/ng_netbase.h"
+#include "net/ng_protnum.h"
+#include "thread.h"
+#include "utlist.h"
+
+#include "net/ng_ipv6/nc.h"
+#include "net/ng_ipv6/netif.h"
+
+#include "net/ng_ipv6.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define _MAX_L2_ADDR_LEN    (8U)
+
+static char _ipv6_stack[NG_IPV6_STACK_SIZE];
+static kernel_pid_t _ipv6_pid = KERNEL_PID_UNDEF;
+
+#if ENABLE_DEBUG
+static char addr_str[NG_IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+static void _ipv6_receive(kernel_pid_t iface, ng_pktsnip_t *pkt);
+static void _ipv6_send(ng_pktsnip_t *pkt, bool prep_hdr);
+static void *_ipv6_event_loop(void *args);
+static void _dispatch_ipv6_pkt(ng_nettype_t type, uint32_t demux_ctx,
```

Comment:
Could add a single line of documentation for each function what it is actually doing?